api/envoy/extensions/filters/http/basic_auth/v3/basic_auth.proto

```diff
@@ -34,3 +36,16 @@ message BasicAuth {
   // Reference to https://httpd.apache.org/docs/2.4/programs/htpasswd.html
   config.core.v3.DataSource users = 1 [(udpa.annotations.sensitive) = true];
 }
+
+message BasicAuthPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    // Disable the basic auth filter for this particular vhost or route.
+    // If disabled is specified in multiple per-filter-configs, the most specific one will be used.
+    bool disabled = 1 [(validate.rules).bool.const = true];
```

Comment:
My implementation follows the approach with `disabled` field, because BasicAuth is enabled on all routes by default. What do you suggest?

Reply:
You can check the doc I posted above. It have provided example about how to disable a filter on any route or vh. It a general feature that Envoy provided and should be used preferably.

Reply:
I meant that I read the doc, but I don't understand what you mean, because another option is to make filter disabled on routes by default and enable it selectively, but this approach would break existing configs. "Prefer the general way" - I did not find term "general" in that doc, so I don't know what you mean.

Reply:
<img width="952" alt="截屏2024-04-04 19 12 31" src="https://github.com/envoyproxy/envoy/assets/12389633/c49c4252-6d90-44e1-a508-ff019ee44288">

Reply:
This is exactly what I implemented.

Reply:
This is a general feature of Envoy to disable a filter in specific route. And all filter could use it directly. The config `envoy.config.route.v3.FilterConfig` is a common wrapper.

You needn't to add a new `disabled` field in your filter's config.

Reply:
Okay... Now I understand it and will remove this field.

Reply:
Check this doc, you can get more detail about the wrapper and the only one drawback of this mechanism https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-filterconfig